awscli/customizations/s3/filegenerator.py

```diff
@@ -157,7 +159,7 @@ def list_files(self, path, dir_op):
         if not self.should_ignore_file(path):
             if not dir_op:
                 size, last_update = get_file_stat(path)
-                yield path, size, last_update
+                yield path, size, last_update, None
```

Comment:
Returning tuples with four elements, some of which are `None` seems like a code smell. Perhaps this method (and other similar methods) should return a dict or a specific object instead of a long tuple?